drivers/bestfortress.c

```diff
@@ -301,23 +356,29 @@ void upsdrv_updateinfo(void)
 		else if (len != 80)
 			checksum_ok = 0;
 		if (checksum_ok) break;
+
+		upsdebugx(1, "%s: failed to read status try %d",
+			  __func__, retry);
 		sleep(SER_WAIT_SEC);
 	}
 
 	if (!p || len < 1 || checksum_ok < 0) {
-		upsdebugx(2, "pointer to data not initialized after processing");
+		/* \todo: Analyze/fix code and rewrite message. */
+		upsdebugx(2, "%s: pointer to data not initialized after processing",
+			__func__);
 		dstate_datastale();
 		return;
 	}
 
 	if (!checksum_ok) {
-		upsdebugx(2, "checksum corruption");
+		upsdebugx(2, "%s: checksum corruption", __func__);
 		upsdebug_hex(3, "buffer", temp, (size_t)len);
 		dstate_datastale();
 		return;
 	}
 
-	/* upslogx(LOG_INFO, "updateinfo: %s", p); */
+	/* Log assuming ASCII which it 99% is.  \todo Improve. */
+	upsdebugx(3, "%s: %s", __func__, p);
```

Comment:
Do we surely expect this is a zero-byte terminated string at least, or should impose a sanity check (or some snprintf) above? The "99%" makes me uneasy :)

Reply:
It is zero-terminated and it is ASCII every time I look at it.  The worst I'd expect is an embedded \r.
But, I want to later add escapedascii to do this right in general.  I can reword the comment.

Reply:
Also I can force-terminate the string to take that concern off the table.

Reply:
If you can force it (or if some `calloc()` already does), could be nice to be on the safe side.

Reply:
So after some looking around, I guess it can be assumed zero-terminated, with `temp[2]==0` and/or `strlen(temp+2)` not being an infinite loop :) and `p` being either `temp` or `temp+2`.